src/excmds.ts

```diff
@@ -2519,14 +2519,22 @@ export async function tabclose(...indexes: string[]) {
         }
         return idFromIndex(id)
     }
+    let ids
     if (indexes.length > 0) {
-        const ids = await Promise.all(indexes.map(index => maybeWinTabToTabId(index)))
-        done = browser.tabs.remove(ids)
+        // Request to close multiple tabs
+        ids = await Promise.all(indexes.map(index => maybeWinTabToTabId(index)))
     } else {
-        // Close current tab
-        done = browser.tabs.remove(await activeTabId())
+        // Request to close the current tab
+        ids = [await activeTabId()]
+    }
+    const tabclosepinned = (await config.getAsync("tabclosepinned") === "true")
```

Comment:
You only need to use getAsync if your function is likely to be called within a few milliseconds of Tridactyl loading in the tab - get is fine otherwise

Reply:
you never know :D sometimes I open and close tabs really fast